platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/FlatLFCustoms.java

```diff
@@ -120,6 +129,34 @@ public Object[] createApplicationSpecificKeysAndValues() {
             "Table.ancestorInputMap.RightToLeft", new LazyModifyInputMap( "Table.ancestorInputMap.RightToLeft", removeCtrlPageUpDownKeyBindings ), // NOI18N
             "Tree.focusInputMap", new LazyModifyInputMap( "Tree.focusInputMap", removeCtrlPageUpDownKeyBindings ), // NOI18N
         };
+        List<Object> result = new ArrayList<>();
+        result.addAll(Arrays.asList(constants));
+        if (Utilities.isWindows()) {
+          /* Make sure button labels appear vertically centered on Windows. On the standard
+          Windows LAF, WindowsButtonUI/WindowsRadioButtonUI/WindowsToggleButtonUI.getPreferredSize
+          add one pixel to the button's height to ensure that it is odd-numbered. This makes the
+          text centered with either Tahoma 11 (the default Swing Windows LAF font) or Segoe UI 12
+          (the default font on modern Windows versions, and on FlatLAF on Windows). */
+          for (String key : new String[] { "Button", "RadioButton", "ToggleButton" }) {
+            UIDefaults defaults = UIManager.getDefaults();
+            Font font = defaults.getFont(key + ".font");
+            Insets bm = defaults.getInsets(key + ".margin");
+            if (font != null && bm != null &&
+                font.getFamily().equals("Segoe UI") && font.getSize() == 12)
+            {
+              if (bm.top == bm.bottom) {
+                result.add(key + ".margin");
+                /* Create an InsetsUIResource rather than an Insets instance if necessary, as
+                FlatLAF treats them differently. Not doing this caused buttons in the main toolbar
+                to become very wide. */
+                result.add(bm instanceof InsetsUIResource
```

Comment:
My inclination is that this should only ever be InsetsUIResource.  Are there instances of plain Insets?  If so, from where, and should they be replaced?

Reply:
It was just a precaution, not something I observed. But since FlatLAF treats Insets and InsetsUIResource differently, I'd imagine cases where FlatLAF customizations might pick one or the other (though this would be very weird). I figured that leaving the same kind of instance was the least likely to cause problems.

Reply:
I meant more that we should probably only replace it at all if it is already a `UIResource`.

Reply:
Oh, sure, that makes sense. Will change it.